test/parallel/test-repl-require-self-referential.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const common = require('../common');
+const fixtures = require('../common/fixtures');
+const assert = require('assert');
+const net = require('net');
+
+if (!common.isMainThread)
+  common.skip('process.chdir is not available in Workers');
+
+const selfRefModule = fixtures.path('self_ref_module');
+process.chdir(selfRefModule);
+const repl = require('repl');
+
+{
+  const server = net.createServer((conn) => {
+    repl.start('', conn).on('exit', () => {
+      conn.destroy();
+      server.close();
+    });
+  });
+
+  const host = common.localhostIPv4;
+  const port = 0;
+  const options = { host, port };
+
+  let answer = '';
+  server.listen(options, function() {
+    options.port = this.address().port;
+    const conn = net.connect(options);
+    conn.setEncoding('utf8');
+    conn.on('data', (data) => answer += data);
+    conn.write('require("self_ref")\n.exit\n');
+  });
```

Comment:
It seems a little overkill to have a whole server here just to test the repl. Can this file not just directly `require('self_ref')`?

Reply:
Yeah, you're right, I can just use a child process as already done in other tests for `require` in the reply in interactive mode.